ksql-cli/src/main/java/io/confluent/ksql/cli/console/Console.java

```diff
@@ -213,58 +221,120 @@ private void registerDefaultCommands() {
   }
 
 
-  private void printTable(
-      List<String> columnHeaders,
-      List<List<String>> rowValues,
-      List<String> header,
-      List<String> footer
-  ) {
+  private static class Table {
+    private final List<String> columnHeaders;
+    private final List<List<String>> rowValues;
+    private final List<String> header;
+    private final List<String> footer;
+
+    public Table(List<String> columnHeaders, List<List<String>> rowValues, List<String> header,
+                 List<String> footer) {
+      this.columnHeaders = columnHeaders;
+      this.rowValues = rowValues;
+      this.header = header;
+      this.footer = footer;
+    }
+
+    public Table(List<String> columnHeaders, List<List<String>> rowValues) {
+      this(columnHeaders, rowValues, Collections.emptyList(), Collections.emptyList());
+    }
+
+    public static class Builder {
+      private final List<String> columnHeaders = new LinkedList<>();
+      private final List<List<String>> rowValues = new LinkedList<>();
+      private final List<String> header = new LinkedList<>();
+      private final List<String> footer = new LinkedList<>();
+
+      Builder() {}
+
+      public Table build() {
```

Comment:
can probably make all the methods package-private - though i'm not overly bothered